sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -573,6 +576,29 @@ public static TypedRead<TableRow> readTableRowsWithSchema() {
             BigQueryUtils.tableRowFromBeamRow());
   }
 
+  @VisibleForTesting
+  static class GenericDatumTransformer<T> extends GenericDatumReader<T> {
```

Comment:
can this extend `DatumReader` rather than `GenericDatumReader`?  You're not actually using any of the GenericDatumReader functionality here.